sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -563,7 +573,17 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   private[this] def castToFloat(from: DataType): Any => Any = from match {
     case StringType =>
       buildCast[UTF8String](_, s => try s.toString.toFloat catch {
-        case _: NumberFormatException => null
+        case _: NumberFormatException =>
+          val str = s.trim.toString
+          if (str.equalsIgnoreCase("infinity")) {
+            Float.PositiveInfinity
+          } else if (str.equalsIgnoreCase("-infinity")) {
+            Float.NegativeInfinity
+          } else if (str.equalsIgnoreCase("nan")) {
+            Float.NaN
+          } else {
+            null
+          }
```

Comment:
To avoid duplications, shall we make a function for this since we have the same `Float` logic twice here and line 1268?

Reply:
@dongjoon-hyun The assignment variable names are different ? Is there an example i can follow where codegen and non-codegen code is refactored to a function ?

Reply:
We don't need an assignment if you handle everything inside the function.
The function will return the right-side value of the assignment.

Reply:
@dongjoon-hyun Ah... thanks .. will try.